### PR TITLE
chore: add isort as pre-commit hook

### PR DESCRIPTION
**Pull request by Mustaballer**
**What kind of change does this PR introduce?**

Adds pre-commit hooks namely [isort](https://github.com/PyCQA/isort), [check-yaml](https://github.com/pre-commit/pre-commit-hooks#check-yaml), [end-of-file-fixer](https://github.com/pre-commit/pre-commit-hooks#end-of-file-fixer), [trailing-whitespace](https://github.com/pre-commit/pre-commit-hooks#trailing-whitespace).

**Summary**

At the moment, OpenAdapt does not provide an automated way to sort imports. However, we have incorporated the isort pre-commit hook, which detects unsorted imports and rejects a commit if they are found. If any unsorted imports are detected, isort will automatically modify the affected files to sort the imports correctly. This helps maintain consistent import organization across the codebase. The other hooks I added I believe are useful and are listed below.

> [check-yaml](https://github.com/pre-commit/pre-commit-hooks#check-yaml): Validates the syntax and structure of YAML files.
> [end-of-file-fixer](https://github.com/pre-commit/pre-commit-hooks#end-of-file-fixer): Ensures that files end with a newline character.
> [trailing-whitespace](https://github.com/pre-commit/pre-commit-hooks#trailing-whitespace): Detects and removes trailing whitespace at the end of lines.
> [isort](https://github.com/PyCQA/isort): Sorts Python import statements in a consistent and standardized manner.

**Checklist**
* [x] My code follows the style guidelines of OpenAdapt
* [x] I have perfomed a self-review of my code
* [x] If applicable, I have added tests to prove my fix is functional/effective
* [x] I have linted my code locally prior to submission
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [x] New and existing unit tests pass locally with my changes

**How can your code be run and tested?**

1. `poetry install`
2. `pre-commit install` note: in root directory
3. Make a commit making changes to some files(only staged files with changes will be executed with the hooks)

### CODE REVIEW

**Comment by Mustaballer** [comment]
Ready for Review :)

**Comment by KrishPatel13** [comment]
There are no newlines after docstrings. 

Should we now move on with this approach now @abrichr ?

(Reminded from: https://github.com/OpenAdaptAI/OpenAdapt/pull/128#discussion_r1206096968)

File: README.md [hunk 1]
```diff
@@ -226,6 +226,29 @@ In summary (from https://stackoverflow.com/a/69673312):
 alembic revision --autogenerate -m "<msg>"
 ```
 
+### Pre-commit Hooks
+
+To ensure code quality and consistency, OpenAdapt uses pre-commit hooks. These hooks will be executed automatically before each commit to perform various checks and validations on your codebase.
```
Comment (abrichr) [inline README.md:231]: Can we also add a pre-commit hook to split long lines like this one onto multiple lines? 😅

File: README.md [hunk 2]
```diff
@@ -226,6 +226,29 @@ In summary (from https://stackoverflow.com/a/69673312):
...
+- [check-yaml](https://github.com/pre-commit/pre-commit-hooks#check-yaml): Validates the syntax and structure of YAML files.
+- [end-of-file-fixer](https://github.com/pre-commit/pre-commit-hooks#end-of-file-fixer): Ensures that files end with a newline character.
+- [trailing-whitespace](https://github.com/pre-commit/pre-commit-hooks#trailing-whitespace): Detects and removes trailing whitespace at the end of lines.
+- [isort](https://github.com/PyCQA/isort): Sorts Python import statements in a consistent and standardized manner.
+
+To set up the pre-commit hooks, follow these steps:
+
+1. Navigate to the root directory of your OpenAdapt repository.
+
+3. Run the following command to install the hooks:
```
Comment (abrichr) [inline README.md:244]: Is there a step 2 that is missing, or is this a typo?

**Review by abrichr — CHANGES_REQUESTED** [review]
Thanks @Mustaballer ! Just a few small things 🙏

File: README.md (hunk 2 above)
Comment (Mustaballer) [reply README.md:244]: typo 👍

File: README.md (hunk 1 above)
Comment (Mustaballer) [reply README.md:231]: Unfortunately there is no pre-commit hook that does this from what I researched. @abrichr I was also considering adding [ruff](https://github.com/astral-sh/ruff)(really fast linter) as a pre-commit hook.

File: README.md (hunk 2 above)
Comment (Mustaballer) [reply README.md]: resolved in https://github.com/OpenAdaptAI/OpenAdapt/pull/319/commits/7099d7f9dd221681b3fd16df65bf22cb89d1c6f8

File: README.md (hunk 1 above)
Comment (abrichr) [reply README.md]: Added a new issue for this: https://github.com/OpenAdaptAI/OpenAdapt/issues/new?assignees=&labels=enhancement&projects=&template=new_feature.yml

**Pull request merged**